File: coding-test/programmers/src/kr/co/programmers/practice/FindPrimeNumber.java
package kr.co.programmers.practice;

public class FindPrimeNumber {

    public static void main(String[] args) {
        int n = 10;
        System.out.println(new Solution().solution(n));
    }

    static class Solution {
        public int solution(int n) {
            int answer = 0;

            boolean[] isNotPrimes = new boolean[n+1];

            for(int i = 2; i <= n; i++){
                if(isNotPrimes[i]){
                    continue;
                }

                answer++;

                for(int j = i * 2; j <= n; j = j + i){
                    isNotPrimes[j] = true;
                }
            }

            return answer;
        }
    }
}

File: coding-test/javascript/src/practice/find-prime-number.js
// @ts-check

/**
 * @param {number} n
 * @returns {number}
 */
function solution(n) {
  let answer = 0

  let isNotPrimes = []
  for (let i = 2; i <= n; i += 1) {
    isNotPrimes[i] = 0
  }

  for (let i = 2; i <= n; i += 1) {
    if (isNotPrimes[i]) {
      continue
    }

    answer += 1

    for (let j = i * 2; j <= n; j += i) {
      isNotPrimes[j] = 1
    }
  }

  return answer
}

console.log(solution(10))
